cps/src/main/java/org/jenkinsci/plugins/workflow/cps/FlowHead.java

```diff
@@ -119,7 +120,9 @@ void setNewHead(FlowNode v) {
             } else {
                 // in recovering from error and such situation, we sometimes need to grow the graph
                 // without running the program.
-                execution.notifyListeners(v);
+                // TODO can CpsThreadGroup.notifyNewHead be used instead?
+                execution.notifyListeners(Collections.singletonList(v), true);
+                execution.notifyListeners(Collections.singletonList(v), false);
```

Comment:
This caller is rarely used, only for error handling. We need to notify _all_ listeners.